app/main.py

```diff
@@ -1,2 +1,117 @@
 class Dictionary:
-    pass
+
+    def __init__(self):
+        self.capacity = 8
+        self.threshold = int(self.capacity * (2 / 3))
+        self.length = 0
+        self.base_data = [None for i in range(self.capacity)]
+
+    def __setitem__(self, key, value):
+        a = 0
+        if self.length >= self.threshold:
+            self.resize()
+
+        index_ = int(hash(key) % self.capacity)
+        if self.base_data[index_] is None:
+            self.base_data[index_] = [key, hash(key), value]
+
+        elif self.base_data[index_][0] == key:
+            self.base_data[index_] = [key, hash(key), value]
+            self.length -= 1
+        else:
+
+            while self.base_data[index_] is not None:
+
+                if self.base_data[index_][0] == key:
+                    self.base_data[index_] = [key, hash(key), value]
+                    self.length -= 1
+                    a = 1
+                if index_ == self.capacity - 1:
+                    index_ = -1
+
+                index_ += 1
+            if a != 1:
+                self.base_data[index_] = [key, hash(key), value]
+        self.length += 1
+
+    def __getitem__(self, key):
+        if self.length != 0:
+            index_ = hash(key) % self.capacity
+            while self.base_data[index_] is not None\
+                    and (index_ < self.capacity):
+                if self.base_data[index_][0] == key:
+                    return self.base_data[index_][2]
+
+                index_ += 1
+
+            raise KeyError(f"Not Key {key} ")
+        else:
+            raise KeyError(f"Not Key {key} ")
+
+    def __len__(self):
+        return int(self.length)
+
+    def resize(self):
+        self.capacity *= 2
+        self.threshold = int(self.capacity * (2 / 3))
+        data_ = []
+        for num in self.base_data:
+            if num is not None:
+                data_.append(num)
+
+        self.base_data = [
+            None for i in range(self.capacity)]
+        self.length = 0
+        for number in data_:
+            self.__setitem__(number[0], number[2])
+
+    def clear(self):
+        self.base_data = [
+            None for i in range(self.capacity)]
+
+    def __delitem__(self, item):
+        for i in range(self.length):
+            if self.base_data[i][2] == item:
+                self.base_data[i] = None
```

Comment:
add these tests

```
@pytest.mark.timeout(5)
def test_deletion():
    items = [(f"Element {i}", i) for i in range(1000)]
    dictionary = Dictionary()
    for key, value in items:
        dictionary[key] = value
    for key, value in items:
        assert dictionary[key] == value
    assert len(dictionary) == len(items)
    for key, value in items:
        del dictionary[key]
    print(len(dictionary))
```

Reply:
I don`t know where exactly to add this test, a separate file? In the condition it is also recommended to check the correctness of the code using the class, how to implement it?

Reply:
add these code to the tests.py file